framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -1371,7 +1375,20 @@ public TransferResult<V, S> visitNarrowingConversion(
     public TransferResult<V, S> visitStringConversion(
             StringConversionNode n, TransferInput<V, S> p) {
         TransferResult<V, S> result = super.visitStringConversion(n, p);
-        result.setResultValue(p.getValueOfSubNode(n.getOperand()));
+
+        Set<AnnotationMirror> operandAnnos = p.getValueOfSubNode(n.getOperand()).getAnnotations();
+        if (operandAnnos.isEmpty() && n.getOperand().getType().getKind() == TypeKind.TYPEVAR) {
+            // use effective annotation if we didn't find any annotations in the value of
```

Comment:
This would be the first call to `getEffectiveAnnotations` in this class. I am wondering how refinement for type variables works elsewhere, e.g. when a TV is assigned to a non-TV local variable. The type checkers seems to do the right thing, but when looking at the CFG, the type of the non-TV variable is actually refined to the TV, which is similarly confusing. I played with something like this:

````java
import org.checkerframework.checker.nullness.qual.Nullable;
class Demo<T extends Object> {
  void foo(T p) {
    @Nullable Object nbl = p;
    nbl.toString();
  }
}
````

So maybe we should improve handling of TVs more generally. Can you look a bit at this and file a separate issue, if you agree this looks odd?

Please improve the comment a bit and make it a full sentence.